### PR TITLE
iscsi: if port is not provided, use default 3260 for target portal

### LINKED ISSUE #15501: iSCSI targetPortal's port is not optional
Documentation says that port number for `targetPortal` is optional, but I've found that not to be true. It's due to the way it detects whether the disk is attached or not.

`targetPortal` also must be an IP.

Refer to:
https://github.com/kubernetes/kubernetes/tree/4da423c41d5bdee9605f4ac53805449681746f84/examples/iscsi

https://github.com/kubernetes/kubernetes/blob/4da423c41d5bdee9605f4ac53805449681746f84/pkg/volume/iscsi/iscsi_util.go#L82

### PR DESCRIPTION
**Pull request by rootfs**
this is to fix #15501

### CODE REVIEW

**Comment by ncdc** [comment]
@kubernetes/rh-storage

File: pkg/volume/iscsi/iscsi.go [hunk 1]
```diff
@@ -94,12 +95,16 @@ func (plugin *iscsiPlugin) newBuilderInternal(spec *volume.Spec, podUID types.UI
 	}
 
 	lun := strconv.Itoa(iscsi.Lun)
+	portal := iscsi.TargetPortal
+	if !strings.Contains(portal, ":") {
+		portal = iscsi.TargetPortal + ":3260"
+	}
```
Comment (markturansky) [inline pkg/volume/iscsi/iscsi.go]: unit test?

**Comment by rootfs** [comment]
@markturansky unit added, also tested against iscsi target

**Comment by markturansky** [comment]
Thanks @rootfs.  LGTM.

@kubernetes/rh-storage one more LGTM for Huamin so I can add the label?

**Comment by childsb** [comment]
LGTM +1

**Comment by markturansky** [comment]
@k8s-bot test this pleast

**Comment by markturansky** [comment]
Shippable Go 1.4 OK.  Flake in Go 1.3 - `F1014 18:05:16.289514   18629 integration.go:918] FAILED: pod never started running timed out waiting for the condition`

**Pull request merged**